posts/2023-11-17-tictactoe-tutorial.mdx

```diff
@@ -0,0 +1,665 @@
+
+```
+npm install short-unique-id copy-to-clipboard
+```
+
+`short-unique-id` will be useful in generating a random unique id for the game room. `copy-to-clipboard` will help us implement a button that can copy the room ID to a user's clipboard.
+
+### Step 4 : Importing the SDK and initialising the providers
+
+Inside the `/pages` folder, you can find `_app.js` file, which is the starting point of a NextJS app. We will be initialising our provider here so that the Waku provider will wrap up the entire application.
+
+So, import the following on top of your file :
+
+```js
+import { LightNodeProvider } from "@waku/react";
+import { Protocols } from "@waku/sdk";
+```
+
+Now, wrap the components with the `<LightNodeProvider>` as follows : 
+
+```js
+    <LightNodeProvider
+      options={{ defaultBootstrap: true }}
+      protocols={[Protocols.Store, Protocols.Filter, Protocols.LightPush]}
+    >
+      <Component {...pageProps} />
+    </LightNodeProvider>
+```
+
+This provider allows the usage of **Store, Filter and LightPush protocols** which will be helpful in implementing all the required functionalities of our game.
+
+### Step 5 : Modify the index.js file
+
+The `index.js` file inside the `/pages` folder renders the index route of the application. Let's make it display our components Header and Hero respectively,
+
+```js 
+import Head from 'next/head'
+import Hero from '@/components/Hero'
+import Header from '@/components/Header'
+
+
+export default function Home() {
+  
+  return (
+    <>
+      <Head>
+        <title>Tic Tac Toe</title>
+        <meta name="description" content="Tic Tac Toe game created with Waku" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main className='bg-black h-screen'>
+        <Header />
+        <Hero />
+      </main>
+    </>
+  )
+}
+
+```
+
+### Step 6 : Header component and Hero component
+
+The header component will carry a logo on the left and a button to our github on the right end :
+
+**Header.js**
+
+```js
+import Link from 'next/link'
+import React from 'react'
+
+export default function Header() {
+    return (
+        <div className='flex p-5 justify-between items-center'>
+            <Link href="/">
+                <div id='logo'>
+                    Your-Logo
+                </div>
+            </Link>
+            <a href='https://github.com/hackyguru/tictactoe' className='text-white flex space-x-3 items-center cursor-pointer'>
+                <p>GitHub</p>
+            </a>
+        </div>
+    )
+}
+```
+
+Remember to replace 'Your-Logo' with your application's logo.
+
+**Hero.js**
+
+The Hero component will allow users to create or join a new game. When a user creates a new game, a unique room ID is generated using the`short-unique-id` package. Where as if the user wants to join an existing game, he can enter the room ID to join in...
```

Comment:
Let's not assum user's gender:D

```suggestion
The Hero component will allow users to create or join a new game. When a user creates a new game, a unique room ID is generated using the`short-unique-id` package. In case users want to join an existing game, they can enter the room ID to join in.
```